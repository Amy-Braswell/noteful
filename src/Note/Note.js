import React from 'react'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { format } from 'date-fns'
import ApiContext from '../ApiContext'
import config from '../config'
import './Note.css'

export default class Note extends React.Component {

  static defaultProps ={
    onDeleteNote: () => {},
  }
  static contextType = ApiContext


  handleClickDelete = () => {
    const noteId = this.props.id
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
    }}
    const url = (`${config.API_ENDPOINT}/notes/${noteId}`)
      fetch(url, options)
      .then(res => {
        if (!res.ok) {
          throw new Error('Something went wrong')
        }
        return res.json
      })
      .then(() => {
        this.context.handleDeleteNote(noteId)
        // allow parent to perform extra behaviour
        this.props.onDeleteNote(noteId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  
  render() {
    const { name, id, modified} = this.props
    if (!name || !id || !modified) {
      return null
    } 
    const formatDate = format(new Date(modified), 'MMMM do yyyy')
    return (
      <div className='Note'>
        <h2 className='Note__title'>
          <Link to={`/notes/${id}`}>
            {name}
          </Link>
        </h2>
        <button
          className='Note__delete'
          type='button'
          onClick={this.handleClickDelete}
        >
          <FontAwesomeIcon icon={['fa', 'trash-alt']} />
          {' '}
          remove
        </button>
        <div className='Note__dates'>
          <div className='Note__dates-modified'>
            Modified
            {' '}
            <span className='Date'>
            {formatDate}
            </span>
          </div>
        </div>
      </div>
    )   
  }
}

