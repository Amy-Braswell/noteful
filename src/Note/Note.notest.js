import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Note from './Note'
import { Link } from 'react-router-dom'
import { format } from 'date-fns'

  //Smoke Test
  it('renders without crashing', () => {
    // start by creating an element to render the component into
    const div = document.createElement('div')
    ReactDOM.render(<Note />, div)
    // Render the component -- Line 9 is the actual test
    // If something is wrong it will fail here.
    ReactDOM.unmountComponentAtNode(div)
  });

describe(`Note component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .Note by default', () => {
    const tree = renderer
    .create(<Note />)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the Note given props', () => {
    const tree = renderer  
    .create(<Note {...props} />)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })
  
})
