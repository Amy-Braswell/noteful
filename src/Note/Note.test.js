import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Note from './Note'
import { Link } from 'react-router-dom'
import { format } from 'date-fns'


  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Note />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

describe(`Note component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .Note by default', () => {
    const tree = renderer
    .create(<Note />)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the Note given props', () => {
    const tree = renderer  
    .create(<Note {...props} />)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })
  
})
