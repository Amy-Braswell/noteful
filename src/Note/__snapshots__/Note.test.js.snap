// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders a .Note by default 1`] = `null`;
